MariaDB: RENAME TABLE is not a supported statement

```RENAME TABLE table TO ...``` is missing for MariaDB. Unfortunately ```ALTER TABLE table RENAME TO ...``` does not work either.

https://mariadb.com/kb/en/rename-table/

**Steps to reproduce:**

1. Add SQL script to rename a table. For example, ```RENAME TABLE IF EXISTS xyz TO zyx;```
2. Parse the SQL... ```Statements statements = CCJSqlParserUtil.parseStatements(sqlString);```

Exception:
```
Caused by: net.sf.jsqlparser.parser.ParseException: Encountered unexpected token: "RENAME" "RENAME"
    at line 344, column 1.
Was expecting one of:
    "("
    ";"
    "ALTER"
    "BEGIN"
    "CALL"
    "COMMENT"
    "COMMIT"
    "CREATE"
    "DECLARE"
    "DELETE"
    "DESCRIBE"
    "DROP"
    "EXEC"
    "EXECUTE"
    "EXPLAIN"
    "GRANT"
    "INSERT"
    "MERGE"
    "REPLACE"
    "RESET"
    "ROLLBACK"
    "SAVEPOINT"
    "SET"
    "SHOW"
    "TRUNCATE"
    "UPDATE"
    "UPSERT"
    "USE"
    "VALUES"
    "WITH"
    <EOF>
    <K_SELECT>
```

--- a/src/main/java/net/sf/jsqlparser/statement/StatementVisitor.java
+++ b/src/main/java/net/sf/jsqlparser/statement/StatementVisitor.java
@@ -11,6 +11,7 @@ package net.sf.jsqlparser.statement;
 
 import net.sf.jsqlparser.statement.alter.Alter;
 import net.sf.jsqlparser.statement.alter.AlterSession;
+import net.sf.jsqlparser.statement.alter.RenameTableStatement;
 import net.sf.jsqlparser.statement.alter.sequence.AlterSequence;
 import net.sf.jsqlparser.statement.comment.Comment;
 import net.sf.jsqlparser.statement.create.index.CreateIndex;
@@ -111,4 +112,6 @@ public interface StatementVisitor {
     void visit(CreateSynonym createSynonym);
 
     void visit(AlterSession alterSession);
+
+    void visit(RenameTableStatement renameTableStatement);
 }

--- a/src/main/java/net/sf/jsqlparser/statement/StatementVisitorAdapter.java
+++ b/src/main/java/net/sf/jsqlparser/statement/StatementVisitorAdapter.java
@@ -11,6 +11,7 @@ package net.sf.jsqlparser.statement;
 
 import net.sf.jsqlparser.statement.alter.Alter;
 import net.sf.jsqlparser.statement.alter.AlterSession;
+import net.sf.jsqlparser.statement.alter.RenameTableStatement;
 import net.sf.jsqlparser.statement.alter.sequence.AlterSequence;
 import net.sf.jsqlparser.statement.comment.Comment;
 import net.sf.jsqlparser.statement.create.index.CreateIndex;
@@ -209,5 +210,10 @@ public class StatementVisitorAdapter implements StatementVisitor {
     @Override
     public void visit(AlterSession alterSession) {
        //@todo: do something usefull here
+    }
+
+    @Override
+    public void visit(RenameTableStatement renameTableStatement) {
+        //@todo: do something usefull here
     }
 }

--- a/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
@@ -1,0 +1,73 @@
+/*-
+ * #%L
+ * JSQLParser library
+ * %%
+ * Copyright (C) 2004 - 2021 JSQLParser
+ * %%
+ * Dual licensed under GNU LGPL 2.1 or Apache License 2.0
+ * #L%
+ */
+/*
+ * Copyright (C) 2021 JSQLParser.
+ *
+ * This library is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation; either version
+ * 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this library;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA
+ */
+
+package net.sf.jsqlparser.statement.alter;
+
+import java.util.Objects;
+import net.sf.jsqlparser.schema.Table;
+import net.sf.jsqlparser.statement.Statement;
+import net.sf.jsqlparser.statement.StatementVisitor;
+
+/**
+ *
+ * @author are
+ * @see  <a href="https://docs.oracle.com/cd/B19306_01/server.102/b14200/statements_9019.htm">Rename</a>
+ */
+public class RenameTableStatement implements Statement {
+    private final Table oldName;
+    private final Table newName;
+
+    public RenameTableStatement(Table oldName, Table newName) {
+        this.oldName = Objects.requireNonNull(oldName, "The OLD NAME of the Rename Statement must not be null.");
+        this.newName = Objects.requireNonNull(newName, "The NEW NAME of the Rename Statement must not be null.");
+    }
+
+    public Table getNewName() {
+        return newName;
+    }
+
+    public Table getOldName() {
+        return oldName;
+    }
+    
+    @Override
+    public void accept(StatementVisitor statementVisitor) {
+        statementVisitor.visit(this);
+      }
+    
+    public StringBuilder appendTo(StringBuilder builder) {
+        builder.append("RENAME ")
+          .append(oldName)
+          .append(" TO ")
+          .append(newName);
+        
+        return builder;
+    }
+
+    @Override
+    public String toString() {
+        return appendTo(new StringBuilder()).toString();
+    }
+}

--- a/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
@@ -25,7 +25,11 @@
 
 package net.sf.jsqlparser.statement.alter;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Objects;
+import java.util.Set;
 import net.sf.jsqlparser.schema.Table;
 import net.sf.jsqlparser.statement.Statement;
 import net.sf.jsqlparser.statement.StatementVisitor;
@@ -36,20 +40,88 @@ import net.sf.jsqlparser.statement.StatementVisitor;
  * @see  <a href="https://docs.oracle.com/cd/B19306_01/server.102/b14200/statements_9019.htm">Rename</a>
  */
 public class RenameTableStatement implements Statement {
-    private final Table oldName;
-    private final Table newName;
+    private final HashMap<Table, Table> tableNames = new HashMap<>();
+    
+    private boolean usingTableKeyword = false;
+    private boolean usingIfExistsKeyword = false;
+    
+    private String waitDirective = "";
 
     public RenameTableStatement(Table oldName, Table newName) {
-        this.oldName = Objects.requireNonNull(oldName, "The OLD NAME of the Rename Statement must not be null.");
-        this.newName = Objects.requireNonNull(newName, "The NEW NAME of the Rename Statement must not be null.");
+        tableNames.put(
+            Objects.requireNonNull(oldName, "The OLD NAME of the Rename Statement must not be null.")
+            , Objects.requireNonNull(newName, "The NEW NAME of the Rename Statement must not be null.")
+        );
+    }
+    
+    public RenameTableStatement(Table oldName, Table newName, boolean usingTableKeyword, boolean usingIfExistsKeyword, String waitDirective) {
+        tableNames.put(
+            Objects.requireNonNull(oldName, "The OLD NAME of the Rename Statement must not be null.")
+            , Objects.requireNonNull(newName, "The NEW NAME of the Rename Statement must not be null.")
+        );
+        
+        this.usingTableKeyword = usingTableKeyword;
+        this.usingIfExistsKeyword = usingIfExistsKeyword;
+        this.waitDirective = waitDirective;
+    }
+    
+    public void addTableNames(Table oldName, Table newName) {
+        tableNames.put(
+            Objects.requireNonNull(oldName, "The OLD NAME of the Rename Statement must not be null.")
+            , Objects.requireNonNull(newName, "The NEW NAME of the Rename Statement must not be null.")
+        );
+    }
+    
+
+    public boolean isUsingTableKeyword() {
+        return usingTableKeyword;
     }
 
-    public Table getNewName() {
-        return newName;
+    public void setUsingTableKeyword(boolean usingTableKeyword) {
+        this.usingTableKeyword = usingTableKeyword;
+    }
+    
+    public RenameTableStatement withUsingTableKeyword(boolean usingTableKeyword) {
+        this.usingTableKeyword = usingTableKeyword;
+        return this;
     }
 
-    public Table getOldName() {
-        return oldName;
+    public boolean isUsingIfExistsKeyword() {
+        return usingIfExistsKeyword;
+    }
+
+    public void setUsingIfExistsKeyword(boolean usingIfExistsKeyword) {
+        this.usingIfExistsKeyword = usingIfExistsKeyword;
+    }
+    
+    public RenameTableStatement withUsingIfExistsKeyword(boolean usingIfExistsKeyword) {
+        this.usingIfExistsKeyword = usingIfExistsKeyword;
+        return this;
+    }
+
+    public String getWaitDirective() {
+        return waitDirective;
+    }
+
+    public void setWaitDirective(String waitDirective) {
+        this.waitDirective = waitDirective;
+    }
+    
+    public RenameTableStatement withWaitDirective(String waitDirective) {
+        this.waitDirective = waitDirective;
+        return this;
+    }
+
+    public int getTableNamesSize() {
+        return tableNames.size();
+    }
+
+    public boolean isTableNamesEmpty() {
+        return tableNames.isEmpty();
+    }
+
+    public Set<Map.Entry<Table, Table>> getTableNames() {
+        return tableNames.entrySet();
     }
     
     @Override
@@ -58,10 +130,29 @@ public class RenameTableStatement implements Statement {
       }
     
     public StringBuilder appendTo(StringBuilder builder) {
-        builder.append("RENAME ")
-          .append(oldName)
-          .append(" TO ")
-          .append(newName);
+        int i=0;
+        for (Entry<Table, Table> e : tableNames.entrySet()) {
+            if (i==0) {
+                builder
+                  .append("RENAME")
+                  .append(usingTableKeyword ? " TABLE " : " ")     
+                  .append(usingIfExistsKeyword ? " IF EXISTS " : " ")                    
+                  .append(e.getKey())
+                  .append(waitDirective!=null && waitDirective.length()>0 ? " " + waitDirective : "")                   
+                  .append(" TO ")
+                  .append(e.getValue());
+            } else {
+                builder
+                  .append(", ")
+                  .append(e.getKey())
+                  .append(" TO ")
+                  .append(e.getValue());
+            }
+            
+            i++;
+        }
+        
+        
         
         return builder;
     }

--- a/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
@@ -10,7 +10,7 @@
 
 package net.sf.jsqlparser.statement.alter;
 
-import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Objects;
@@ -25,7 +25,7 @@ import net.sf.jsqlparser.statement.StatementVisitor;
  * @see  <a href="https://docs.oracle.com/cd/B19306_01/server.102/b14200/statements_9019.htm">Rename</a>
  */
 public class RenameTableStatement implements Statement {
-    private final HashMap<Table, Table> tableNames = new HashMap<>();
+    private final LinkedHashMap<Table, Table> tableNames = new LinkedHashMap<>();
     
     private boolean usingTableKeyword = false;
     private boolean usingIfExistsKeyword = false;

--- a/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/RenameTableStatement.java
@@ -7,21 +7,6 @@
  * Dual licensed under GNU LGPL 2.1 or Apache License 2.0
  * #L%
  */
-/*
- * Copyright (C) 2021 JSQLParser.
- *
- * This library is free software; you can redistribute it and/or modify it under the terms of the
- * GNU Lesser General Public License as published by the Free Software Foundation; either version
- * 2.1 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
- * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License along with this library;
- * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
- * 02110-1301 USA
- */
 
 package net.sf.jsqlparser.statement.alter;
 
@@ -151,9 +136,6 @@ public class RenameTableStatement implements Statement {
             
             i++;
         }
-        
-        
-        
         return builder;
     }
 

--- a/src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java
+++ b/src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java
@@ -11,6 +11,7 @@ package net.sf.jsqlparser.util;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import net.sf.jsqlparser.expression.AnalyticExpression;
 import net.sf.jsqlparser.expression.AnyComparisonExpression;
@@ -1028,7 +1029,9 @@ public class TablesNamesFinder implements SelectVisitor, FromItemVisitor, Expres
 
     @Override
     public void visit(RenameTableStatement renameTableStatement) {
-        renameTableStatement.getOldName().accept(this);
-        renameTableStatement.getNewName().accept(this);
+        for (Map.Entry<Table, Table> e : renameTableStatement.getTableNames()) {
+            e.getKey().accept(this);
+            e.getValue().accept(this);
+          }
     }
 }

--- a/src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java
+++ b/src/main/java/net/sf/jsqlparser/util/TablesNamesFinder.java
@@ -68,6 +68,7 @@ import net.sf.jsqlparser.schema.Table;
 import net.sf.jsqlparser.statement.*;
 import net.sf.jsqlparser.statement.alter.Alter;
 import net.sf.jsqlparser.statement.alter.AlterSession;
+import net.sf.jsqlparser.statement.alter.RenameTableStatement;
 import net.sf.jsqlparser.statement.alter.sequence.AlterSequence;
 import net.sf.jsqlparser.statement.comment.Comment;
 import net.sf.jsqlparser.statement.create.index.CreateIndex;
@@ -1023,5 +1024,11 @@ public class TablesNamesFinder implements SelectVisitor, FromItemVisitor, Expres
         for (JsonFunctionExpression expr: expression.getExpressions()) {
             expr.getExpression().accept(this);
         }
+    }
+
+    @Override
+    public void visit(RenameTableStatement renameTableStatement) {
+        renameTableStatement.getOldName().accept(this);
+        renameTableStatement.getNewName().accept(this);
     }
 }

--- a/src/main/java/net/sf/jsqlparser/util/deparser/StatementDeParser.java
+++ b/src/main/java/net/sf/jsqlparser/util/deparser/StatementDeParser.java
@@ -30,6 +30,7 @@ import net.sf.jsqlparser.statement.Statements;
 import net.sf.jsqlparser.statement.UseStatement;
 import net.sf.jsqlparser.statement.alter.Alter;
 import net.sf.jsqlparser.statement.alter.AlterSession;
+import net.sf.jsqlparser.statement.alter.RenameTableStatement;
 import net.sf.jsqlparser.statement.alter.sequence.AlterSequence;
 import net.sf.jsqlparser.statement.comment.Comment;
 import net.sf.jsqlparser.statement.create.index.CreateIndex;
@@ -348,5 +349,10 @@ public class StatementDeParser extends AbstractDeParser<Statement> implements St
     @Override
     public void visit(AlterSession alterSession) {
         new AlterSessionDeParser(buffer).deParse(alterSession);
+    }
+
+    @Override
+    public void visit(RenameTableStatement renameTableStatement) {
+        renameTableStatement.appendTo(buffer);
     }
 }

--- a/src/main/java/net/sf/jsqlparser/util/validation/validator/StatementValidator.java
+++ b/src/main/java/net/sf/jsqlparser/util/validation/validator/StatementValidator.java
@@ -28,6 +28,7 @@ import net.sf.jsqlparser.statement.Statements;
 import net.sf.jsqlparser.statement.UseStatement;
 import net.sf.jsqlparser.statement.alter.Alter;
 import net.sf.jsqlparser.statement.alter.AlterSession;
+import net.sf.jsqlparser.statement.alter.RenameTableStatement;
 import net.sf.jsqlparser.statement.alter.sequence.AlterSequence;
 import net.sf.jsqlparser.statement.comment.Comment;
 import net.sf.jsqlparser.statement.create.function.CreateFunction;
@@ -273,7 +274,13 @@ public class StatementValidator extends AbstractValidator<Statement> implements 
         //@todo: write something usefull here
     }
     
+    @Override
     public void visit(AlterSession alterSession) {
+        //@todo: write something usefull here
+    }
+
+    @Override
+    public void visit(RenameTableStatement renameTableStatement) {
         //@todo: write something usefull here
     }
 

--- a/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
+++ b/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
@@ -752,17 +752,43 @@ ResetStatement Reset(): {
 }
 
 RenameTableStatement RenameTableStatement(): {
+    RenameTableStatement renameTableStatement;
     Table oldName;
     Table newName;
+    boolean usingTableKeyword=false;
+    boolean usesIfExistsKeyword=false;
+    String waitDirective = "";
+    Token token;
 }
 {
-    <K_RENAME>
+    <K_RENAME> 
+    [ LOOKAHEAD(2) <K_TABLE> { usingTableKeyword = true; } ]
+    [ LOOKAHEAD(2) <K_IF> <K_EXISTS> { usesIfExistsKeyword = true; } ]
     oldName = Table()
+    [ ( 
+        <K_WAIT> token=<S_LONG> { waitDirective = "WAIT " + token.image; }
+        |
+        <K_NOWAIT>  { waitDirective = "NOWAIT"; }
+       ) ]
     <K_TO>
     newName = Table()
 
     { 
-        return new RenameTableStatement(oldName, newName); 
+        renameTableStatement = new RenameTableStatement(oldName, newName, usingTableKeyword, usesIfExistsKeyword, waitDirective); 
+    }
+
+    (
+        ","
+        oldName = Table()
+        <K_TO>
+        newName = Table()
+        {
+            renameTableStatement.addTableNames(oldName, newName);
+        }
+    )*
+
+    { 
+        return renameTableStatement;
     }
 }
 

--- a/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
+++ b/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
@@ -551,6 +551,8 @@ Statement SingleStatement() :
         |
         stm = Set()
         |
+        stm = RenameTableStatement()
+        |
         stm = Reset()
         |
         LOOKAHEAD(ShowColumns())
@@ -747,6 +749,21 @@ ResetStatement Reset(): {
     <K_RESET> ( LOOKAHEAD(2) <K_DATETIMELITERAL> <K_ZONE> {name = "Time Zone"; } | name = RelObjectName() | all = <K_ALL> {name = all.image; } )
     { reset = new ResetStatement(name); }
     { return reset; }
+}
+
+RenameTableStatement RenameTableStatement(): {
+    Table oldName;
+    Table newName;
+}
+{
+    <K_RENAME>
+    oldName = Table()
+    <K_TO>
+    newName = Table()
+
+    { 
+        return new RenameTableStatement(oldName, newName); 
+    }
 }
 
 DescribeStatement Describe(): {

--- a/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
@@ -1,0 +1,100 @@
+/*-
+ * #%L
+ * JSQLParser library
+ * %%
+ * Copyright (C) 2004 - 2021 JSQLParser
+ * %%
+ * Dual licensed under GNU LGPL 2.1 or Apache License 2.0
+ * #L%
+ */
+/*
+ * Copyright (C) 2021 JSQLParser.
+ *
+ * This library is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation; either version
+ * 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this library;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA
+ */
+
+package net.sf.jsqlparser.statement.alter;
+
+import java.util.List;
+import net.sf.jsqlparser.JSQLParserException;
+import net.sf.jsqlparser.parser.CCJSqlParserUtil;
+import net.sf.jsqlparser.statement.Statement;
+import net.sf.jsqlparser.statement.StatementVisitorAdapter;
+import net.sf.jsqlparser.test.TestUtils;
+import net.sf.jsqlparser.util.TablesNamesFinder;
+import net.sf.jsqlparser.util.validation.ValidationTestAsserts;
+import net.sf.jsqlparser.util.validation.feature.DatabaseType;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ *
+ * @author <a href="mailto:andreas@manticore-projects.com">Andreas Reichel</a>
+ */
+public class RenameTableStatementTest {
+  /**
+   * This test will parse and deparse the statement and assures the functional coverage by
+   * JSQLParser.
+   * 
+   * @throws net.sf.jsqlparser.JSQLParserException
+   */
+  @Test
+  public void testStatement() throws JSQLParserException {
+    String sqlStr = "RENAME oldTableName TO newTableName";
+
+    TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+  }
+
+  /**
+   * This test will trigger the method {@link StatementVisitorAdaptor#visit() Visit Method} in the
+   * StatementVisitorAdaptor needed for the Code Coverage.
+   * 
+   * @throws net.sf.jsqlparser.JSQLParserException
+   */
+  @Test
+  public void testStatementVisitorAdaptor() throws JSQLParserException {
+    String sqlStr = "RENAME oldTableName TO newTableName";
+
+    CCJSqlParserUtil.parse(sqlStr).accept(new StatementVisitorAdapter());
+  }
+
+  /**
+   * This test will trigger the method {@link TableNamesFinder#visit() Visit Method} in the
+   * TableNamesFinder needed for the Code Coverage.
+   * 
+   * @throws net.sf.jsqlparser.JSQLParserException
+   */
+  @Test
+  public void testTableNamesFinder() throws JSQLParserException {
+    String sqlStr = "RENAME oldTableName TO newTableName";
+
+    Statement statement = CCJSqlParserUtil.parse(sqlStr);
+    List<String> tables = new TablesNamesFinder().getTableList(statement);
+    Assert.assertEquals(2, tables.size());
+    Assert.assertTrue(tables.contains("oldTableName"));
+    Assert.assertTrue(tables.contains("newTableName"));
+  }
+
+  /**
+   * This test will trigger the method {@link ExpressionValidator#visit() Visit Method} in the
+   * ExpressionValidator needed for the Code Coverage.
+   * 
+   * @throws net.sf.jsqlparser.JSQLParserException
+   */
+  @Test
+  public void testValidator() throws JSQLParserException {
+    String sqlStr = "RENAME oldTableName TO newTableName";
+
+    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
+  }
+}

--- a/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
@@ -7,21 +7,6 @@
  * Dual licensed under GNU LGPL 2.1 or Apache License 2.0
  * #L%
  */
-/*
- * Copyright (C) 2021 JSQLParser.
- *
- * This library is free software; you can redistribute it and/or modify it under the terms of the
- * GNU Lesser General Public License as published by the Free Software Foundation; either version
- * 2.1 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
- * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License along with this library;
- * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
- * 02110-1301 USA
- */
 
 package net.sf.jsqlparser.statement.alter;
 

--- a/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
@@ -51,7 +51,15 @@ public class RenameTableStatementTest {
   @Test
   public void testStatement() throws JSQLParserException {
     String sqlStr = "RENAME oldTableName TO newTableName";
-
+    TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+    
+    sqlStr = "RENAME TABLE old_table TO backup_table, new_table TO old_table";
+    TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+    
+    sqlStr = "RENAME TABLE IF EXISTS old_table WAIT 20 TO backup_table, new_table TO old_table";
+    TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+    
+    sqlStr = "RENAME TABLE IF EXISTS old_table NOWAIT TO backup_table, new_table TO old_table";
     TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
   }
 

--- a/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
@@ -13,6 +13,7 @@ package net.sf.jsqlparser.statement.alter;
 import java.util.List;
 import net.sf.jsqlparser.JSQLParserException;
 import net.sf.jsqlparser.parser.CCJSqlParserUtil;
+import net.sf.jsqlparser.schema.Table;
 import net.sf.jsqlparser.statement.Statement;
 import net.sf.jsqlparser.statement.StatementVisitorAdapter;
 import net.sf.jsqlparser.test.TestUtils;
@@ -90,4 +91,21 @@ public class RenameTableStatementTest {
 
     ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
   }
+  
+  @Test
+  public void testObjectAccess() {
+      Table oldTable = new Table("oldTableName");
+      Table newTable = new Table("newTableName");
+      
+      RenameTableStatement renameTableStatement = new RenameTableStatement(oldTable, newTable);
+      renameTableStatement.withUsingTableKeyword(true).setUsingTableKeyword(false);
+      renameTableStatement.withUsingIfExistsKeyword(true).setUsingIfExistsKeyword(false);
+      renameTableStatement.withWaitDirective("NOWAIT").setWaitDirective("WAIT 20");
+      
+      Assert.assertFalse(renameTableStatement.isTableNamesEmpty());
+      Assert.assertTrue(renameTableStatement.getTableNamesSize()>0);
+      Assert.assertFalse(renameTableStatement.isUsingTableKeyword());
+      Assert.assertFalse(renameTableStatement.isUsingIfExistsKeyword());
+      Assert.assertEquals("WAIT 20", renameTableStatement.getWaitDirective());
+    }
 }

